### ISSUE TITLE
LoRA fine-tuning training script

### ISSUE THREAD

**Issue by timlawrenz** [timlawrenz/DINO-X#22]
## Summary
A standard supervised training script for fine-tuning frozen DINO-X backbones with LoRA adapters.

## CLI
```bash
python scripts/finetune_lora.py \
  --backbone timlawrenz/dinox-ct-vit-small-v1 \
  --task classification \
  --train-data /path/to/train/ \
  --val-data /path/to/val/ \
  --rank 8 \
  --epochs 50 \
  --lr 1e-3 \
  --output my-adapter/
```

## Features
- Downloads backbone from HuggingFace (or loads local checkpoint)
- Applies LoRA to specified layers
- Standard supervised training (classification, regression, segmentation head)
- Early stopping on validation metric
- Saves adapter only (not full model)
- Logs to tensorboard

## Example: LIDC Malignancy
Demonstrate fine-tuning on LIDC malignancy with LoRA rank=8.
Compare: LoRA vs full fine-tune vs frozen linear probe.

## Depends On
- #21 (Adapter save/load)

**Comment by timlawrenz** [comment]
Implemented in scripts/finetune_lora.py (commit 4e86d70). Features: backbone loading from HF/local, LoRA injection via zoo/peft, classification + regression tasks, CSV-based labeled data with optional spacing, cosine LR + warmup, early stopping, macro-F1 metrics, adapter + head save/load. 18 tests in test_finetune_lora.py.
